### PR TITLE
Added Calculation dependencies of reports and dashboards

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30118

## Description
Added dependencies calculation of dashboards and reports

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/common/update_id_set.py
Comment (teizenman) [inline demisto_sdk/commands/common/update_id_set.py:830]: Are you handling classifiers here?

**Review by teizenman — CHANGES_REQUESTED** [review]
You should use the TestSuite for the testing instead of test_data files.
@roysagi correct me if I'm wrong

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:830]: No, was a mistake, deleting it

File: demisto_sdk/commands/common/update_id_set.py
Comment (teizenman) [inline demisto_sdk/commands/common/update_id_set.py:822]: It might be a good idea to use a `set` and not a `list`

**Review by teizenman — CHANGES_REQUESTED** [review]
You can delete the test_data files now right?

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:822]: This is a problem, cause there are functions that are expecting this to be a list (like done in script and etc).

Comment (teizenman) [reply demisto_sdk/commands/common/update_id_set.py:822]: You can use a set at the beginning and then changing it back to list at the end?

**Comment by abaumgarten** [comment]
Build is failing on the error from here: demisto/etc#31320

File: demisto_sdk/commands/common/update_id_set.py
Comment (guykeller) [inline demisto_sdk/commands/common/update_id_set.py:816]: Also, why aren't we giving the variables better name? because it's like that throughout the code?
```suggestion
def parse_dashboard_or_report_data(layouts, data, json_data, path):
```

**Review by guykeller — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:810]: Yes, I thought that it would be the correct way to be consistent

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:816]: Done

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:825]: Done

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:827]: Yes, it seems like in all files (dashboards and reports) , 'query' exists

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:830]: Done

File: demisto_sdk/commands/common/update_id_set.py
Comment (guykeller) [inline demisto_sdk/commands/common/update_id_set.py:803]: WDYT? your call
```suggestion
    dashboard_data = get_json(path)
```

File: demisto_sdk/commands/common/update_id_set.py
Comment (guykeller) [inline demisto_sdk/commands/common/update_id_set.py:809]: same here

**Review by guykeller — APPROVED** [review]
Please fix my last note and wait for feedback from @roysagi regarding the other note

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:803]: Done

File: demisto_sdk/commands/common/update_id_set.py
Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:809]: Done

File: demisto_sdk/commands/common/update_id_set.py
Comment (roysagi) [reply demisto_sdk/commands/common/update_id_set.py:838]: Essentially you are right - we can add a general function to create the `entity_data` that can get all the possible keys and pass all the parsing methods to it -  that is a good idea

Comment (abaumgarten) [reply demisto_sdk/commands/common/update_id_set.py:838]: Great, on it.

**Review by teizenman — APPROVED** [review]

**Pull request merged**
